Library/Homebrew/extend/os/linux/cask/config.rb

```diff
@@ -0,0 +1,50 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "os/linux"
+
+module OS
+  module Linux
+    module Cask
+      module Config
+        extend T::Helpers
+
+        requires_ancestor { ::Cask::Config }
+
+        DEFAULT_DIRS = T.let({
+          vst_plugindir:        "~/.vst",
+          vst3_plugindir:       "~/.vst3",
+          fontdir:              "#{ENV.fetch("XDG_DATA_HOME", "~/.local/share")}/fonts",
+          appdir:               nil,
+          keyboard_layoutdir:   nil,
+          colorpickerdir:       nil,
+          prefpanedir:          nil,
+          qlplugindir:          nil,
+          mdimporterdir:        nil,
+          servicedir:           nil,
+          dictionarydir:        nil,
+          screen_saverdir:      nil,
+          input_methoddir:      nil,
+          internet_plugindir:   nil,
+          audio_unit_plugindir: nil,
+        }.freeze, T::Hash[Symbol, T.nilable(String)])
+
+        sig { returns(T::Hash[Symbol, T.untyped]) }
+        def self.defaults
```

Comment:
@Homebrew/brew I can't get the code to pick up this extended version, no matter what I try. Does anyone have a suggestion how to do that?

Reply:
Ugh, guilty 😬 

`prepend`ing (like `include`) only affects instance methods. If I understand what you're trying to do here, you'll want to write this as an instance method in a submodule (e.g. `ClassMethods`) which is then prepended to the singleton class. See `Keg` for an example of this dual-`prepend` approach, specifically: https://github.com/Homebrew/brew/blob/e23d079af540c6088c4ddc8fa9557e1b79bafe1a/Library/Homebrew/extend/os/mac/keg.rb#L130-L131

(I'm entirely to blame for this convoluted mess, and I'm open to alternatives. It's probably useful to document this for others, maybe in `Typechecking.md`, though it has discoverability issues.)

Reply:
That doesn't look all that bad, I wouldn't have guessed that this was type check related though, so I probably wouldn't have found it in the documentation.

Reply:
I've done this now, but it somehow doesn't get invoked when I run `brew install --cask font-<something>`. And I have no idea why.

Reply:
To (hopefully) ask the obvious, you're running that command on Linux, correct?

Reply:
Yeah, was running it in a codespace.

Reply:
I think this will fix you up (I know it's confusing…): https://github.com/Homebrew/brew/pull/18874/files#r1909348690